src/utils/MzMLSplitter.cpp

```diff
@@ -0,0 +1,206 @@
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Hendrik Weisser $
+// $Authors: Hendrik Weisser $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/APPLICATIONS/TOPPBase.h>
+#include <OpenMS/FORMAT/MzMLFile.h>
+#include <QFile>
+
+using namespace OpenMS;
+using namespace std;
+
+//-------------------------------------------------------------
+// Doxygen docu
+//-------------------------------------------------------------
+
+/**
+    @page UTILS_MzMLSplitter MzMLSplitter
+
+    @brief Splits an mzML file into multiple parts
+
+    This utility will split an input mzML file into @e N parts, with an approximately equal number of spectra and chromatograms in each part. @e N is set by the parameter @p parts; optionally only spectra (parameter @p no_chrom) or only chromatograms (parameter @p no_spec) can be transferred to the output.
+
+    Alternatively to setting the number of parts directly, a target maximum file size for the parts can be specified (parameters @p size and @p unit). The number of parts is then calculated by dividing the original file size by the target and rounding up. Note that the resulting parts may actually be bigger than the target size (due to meta data that is included in every part) or that more parts than necessary may be produced (if spectra or chromatograms are removed via @p no_spec/@p no_chrom).
+
+    This tool cannot be used as part of a TOPPAS workflow, because the number of output files is variable.
+
+    <B>The command line parameters of this tool are:</B>
+    @verbinclude UTILS_MzMLSplitter.cli
+    <B>INI file documentation of this tool:</B>
+    @htmlinclude UTILS_MzMLSplitter.html
+*/
+
+// We do not want this class to show up in the docu:
+/// @cond TOPPCLASSES
+
+class TOPPMzMLSplitter :
+  public TOPPBase
+{
+public:
+
+  TOPPMzMLSplitter() :
+    TOPPBase("MzMLSplitter", "Splits an mzML file into multiple parts", false)
+  {
+  }
+
+protected:
+
+  void registerOptionsAndFlags_()
+  {
+    registerInputFile_("in", "<file>", "", "Input file");
+    setValidFormats_("in", ListUtils::create<String>("mzML"));
+    registerStringOption_("out", "<file>", "", "Prefix for output files ('_part1.mzML' etc. will be appended; default: same as 'in' without the file extension)", false);
+    registerIntOption_("parts", "<num>", 1, "Number of parts to split into (takes precedence over 'size' if set)", false);
+    setMinInt_("parts", 1);
+    registerIntOption_("size", "<num>", 0, "Approximate upper limit for resulting file sizes (in 'unit')", false);
+    setMinInt_("size", 0);
+    registerStringOption_("unit", "<choice>", "MB", "Unit for 'size' (base 1024)", false);
+    setValidStrings_("unit", ListUtils::create<String>("KB,MB,GB"));
+    // @TODO:
+    // registerFlag_("precursor", "Make sure precursor spectra end up in the same part as their fragment spectra");
```

Comment:
I thought about it, but it seems non-trivial to implement well (given that there could be multiple levels of fragmentation or that the data could already be filtered) and I don't see an urgent need for it. I'd just leave it out for now.